js/liqui.js

```diff
@@ -617,21 +628,40 @@ module.exports = class liqui extends Exchange {
         return { 'url': url, 'method': method, 'body': body, 'headers': headers };
     }
 
-    async request (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
-        let response = await this.fetch2 (path, api, method, params, headers, body);
+    handleErrors (httpCode, reason, url, method, headers, body) {
+        if (httpCode !== 200)
+            return; // resort to default error handler
+        let response = undefined;
+        if ((body[0] === '{') || (body[0] === '[')) {
+            response = JSON.parse (body);
+        } else {
+            // if not a JSON response
+            throw new ExchangeError (this.id + ' returned a non-JSON reply: ' + body);
+        }
         if ('success' in response) {
             if (!response['success']) {
-                if (response['error'].indexOf ('Not enougth') >= 0) { // not enougTh is a typo inside Liqui's own API...
-                    throw new InsufficientFunds (this.id + ' ' + this.json (response));
-                } else if (response['error'] == 'Requests too often') {
-                    throw new DDoSProtection (this.id + ' ' + this.json (response));
-                } else if ((response['error'] == 'not available') || (response['error'] == 'external service unavailable')) {
-                    throw new DDoSProtection (this.id + ' ' + this.json (response));
+                const code = response['code'];
+                const message = response['error'];
+                const feedback = this.id + ' ' + this.json (response);
+                const errorCodes = this.errorCodes;
+                if (code in errorCodes) {
+                    const ExceptionRef = errorCodes[code];
```

Comment:
You made it with a low capital letter last time.
I don't like `ExceptionRef` either but can't come up with a better name.
But the low capital letter I strongly dislike too - it's a class, `new exception()` looks quite weird.

Reply:
I don't like the `Ref` as I don't think it is a reference and I'm not very comfortable with abbreviations ) 
Let's call it ExceptionType or ExceptionClass? I think it's the closest meaning. How about that?

Reply:
`ExceptionClass` )
It's a Class, so let's be explicit )

Reply:
> I don't think it is a reference

Why do you think it's not a reference, btw?

Reply:
@mkutny well, I had experience with non-interpreted languages, and it just so happens that in JS the prototypes are actual *instances* that can be referenced indeed, so, despite that term is absolutely correct in JS, it would be semantically misleading later when we cover a larger set of languages.

Reply:
Ok, I came up with a name which should make both you and the linter happy )

Reply:
Yay! )